### PR TITLE
Converted the AppModal to a AppPage and introducing PageComponent.

### PR DESCRIPTION
**Pull request by aldipower**
Now an app is displayed in a page instead of a modal.
#810 
Creating a new app and showing the about data is still done in a modal.

Introducing PageComponent.
Moved components to be rendered into own function.

app

### CODE REVIEW

File: src/main/resources/assets/js/components/modals/AboutModalComponent.jsx [hunk 1]
```diff
@@ -6,14 +6,6 @@ var BackboneMixin = require("../../mixins/BackboneMixin");
 var ModalComponent = require("../components/../ModalComponent");
```
Comment (pyronicide) [inline src/main/resources/assets/js/components/modals/AboutModalComponent.jsx]: Isn't this a page now? I don't see the ModalComponent being used, and it seems the name + directories are wrong.

File: src/main/resources/assets/js/components/modals/AboutModalComponent.jsx [hunk 2]
```diff
@@ -37,9 +29,19 @@ var AboutModalComponent = React.createClass({
     return this.state.info;
   },
 
+  getInfo: function (attr, showUnspecifiedText) {
```
Comment (pyronicide) [inline src/main/resources/assets/js/components/modals/AboutModalComponent.jsx]: Why do we need `showUnspecifiedText` as a special option? Seems like a good default case for the version and then we get rid of a ton of special logic.

File: src/main/resources/assets/js/components/Marathon.jsx [hunk 3]
```diff
@@ -385,46 +385,105 @@ var Marathon = React.createClass({
...
+            destroyDeployment={this.destroyDeployment}
+            fetchState={this.state.deploymentsFetchState} />
+        </TabPaneComponent>
+      </TogglableTabsComponent>
+    );
+    /* jshint trailing:true, quotmark:true, newcap:true */
+    /* jscs:enable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+  },
+
+  getNewAppModal: function () {
```
Comment (pyronicide) [inline src/main/resources/assets/js/components/Marathon.jsx]: I thought this got rid of the app modals.

File: src/main/resources/assets/js/components/modals/AboutModalComponent.jsx (hunk 1 above)
Comment (aldipower) [inline src/main/resources/assets/js/components/modals/AboutModalComponent.jsx]: No, it isn't. I also thought this, but I was wrong.
See:
https://github.com/mesosphere/marathon/pull/1123#issuecomment-72574699

File: src/main/resources/assets/js/components/Marathon.jsx (hunk 3 above)
Comment (aldipower) [inline src/main/resources/assets/js/components/Marathon.jsx]: Same thing as above:
https://github.com/mesosphere/marathon/pull/1123#issuecomment-72574699

File: src/main/resources/assets/js/components/modals/AboutModalComponent.jsx (hunk 2 above)
Comment (aldipower) [inline src/main/resources/assets/js/components/modals/AboutModalComponent.jsx]: Yeah, that makes more sense.

**Comment by aatumanova** [comment]
Nice work @aldipower! I have two design suggestions:

1. The top of the page has a lot of visual "striping" going on with the different gray backgrounds.  I suggest we simplify the design by removing the dark background behind the app name and actions.
2. The back arrow on the right doesn't feel right (I associate the right side with "forward"). Can we implement the arrow on the left, or a breadcrumb?

Mockup:
image

**Comment by aldipower** [comment]
Hi @aatumanova !
Cool.

I've tried it with the breadcrumbs and have some variants to suggest.
Btw, is there a unicode for the large greater arrow you have used?

1.
Putting the task also in the new breadcrumb line, like @mlunoe suggested.
With grey links:
grey-task
With green links, so the user knows that it is clickable:
green-task

2.
Let the old seperated task breadcrumb line where it is:
green-task-tasklist
User has a better control to come back to the task list.

3.
Task overview with grey breadcrumb links.
grey-apps

And still we could use the old close button (X) on the right side? 
Even though it's not a modal? @pyronicide

**Comment by mlunoe** [comment]
@aldipower and @aatumanova I think the bottom suggestion 1 looks best, but the headline "/Sleep30 `Running`" seems a little redundant. Maybe we should remove that, since the app name is already displayed in the breadcrumbs, and then show `Running` in front of the name (in the breadcrumbs) when on the main page of the app. Something along the line like this (Spacing is a bit off, just hacked this together):



Also, I think it would work much better if a single task is displayed on a separate page, meaning only the breadcrumbs and the task info:

And not show the app header, with suspend, scale, destroy app and the "Tasks" and "Configuration" tabs.

Finally, I don't think the (X) on the right is necessary, since you can go to the apps page by clicking the "Apps" breadcrumb or the "Apps" tab in the top.

Does that make sense to you guys?

**Comment by aldipower** [comment]
Ok, @mlunoe and @aatumanova .
What do you think?

bread1
bread2

**Comment by aatumanova** [comment]
Hey guys! "Running" feels like it's part of the breadcrumb in this design. I would suggest separating the breadcrumb from the status and keeping the header app name. I know this creates duplicate content (app name repeats twice), but I think this is the better trade off to a confusing placement for "Running". Here's a design that keeps the breadcrumb content separated from the rest of the page for a cleaner design.

image

The background color behind the breadcrumbs is #2a2d2f, and I had to darken the background color behind the header to #222426.

**Comment by aldipower** [comment]
@aatumanova Yeah, that makes sense.
So here we are:
m1
m2

**Pull request closed without merging**